packages/compass-query-bar/src/components/query-history/query-item/query-item-action-buttons.tsx

```diff
@@ -1,54 +1,85 @@
-import { Icon, IconButton } from '@mongodb-js/compass-components';
+import { Icon, IconButton, css } from '@mongodb-js/compass-components';
 import React from 'react';
 
 type ActionButtonProps = {
-  onClick: () => void;
+  onClick?: () => void;
+  visible?: boolean;
 };
 
-export const FavoriteActionButton = ({ onClick }: ActionButtonProps) => {
+const VISIBLE_DISPLAY = 'inline';
+const INVISIBLE_DISPLAY = 'none';
+
+const exhaustIfInteractive =
+  (onClick: (() => void) | undefined) => (event: React.MouseEvent) => {
+    if (onClick) {
+      event.stopPropagation();
+      onClick?.();
+    }
+  };
+
+export const FavoriteActionButton = ({
+  onClick,
+  visible,
+}: ActionButtonProps) => {
   return (
     <IconButton
+      className={css({
+        display: visible ? VISIBLE_DISPLAY : INVISIBLE_DISPLAY,
+      })}
       data-testid="query-history-button-fav"
       aria-label="Favorite Query"
       title="Favorite Query"
-      onClick={(event) => {
-        event.stopPropagation();
-        onClick();
-      }}
+      onClick={exhaustIfInteractive(onClick)}
     >
       <Icon glyph="Favorite" />
     </IconButton>
   );
 };
 
-export const CopyActionButton = ({ onClick }: ActionButtonProps) => {
+export const CopyActionButton = ({ onClick, visible }: ActionButtonProps) => {
   return (
     <IconButton
+      className={css({
+        display: visible ? VISIBLE_DISPLAY : INVISIBLE_DISPLAY,
+      })}
       data-testid="query-history-button-copy-query"
       aria-label="Copy Query to Clipboard"
       title="Copy Query to Clipboard"
-      onClick={(event) => {
-        event.stopPropagation();
-        onClick();
-      }}
+      onClick={exhaustIfInteractive(onClick)}
     >
       <Icon glyph="Copy" />
     </IconButton>
   );
 };
 
-export const DeleteActionButton = ({ onClick }: ActionButtonProps) => {
+export const DeleteActionButton = ({ onClick, visible }: ActionButtonProps) => {
   return (
     <IconButton
+      className={css({
+        display: visible ? VISIBLE_DISPLAY : INVISIBLE_DISPLAY,
+      })}
       data-testid="query-history-button-delete-recent"
       aria-label="Delete Query from List"
       title="Delete Query from List"
-      onClick={(event) => {
-        event.stopPropagation();
-        onClick();
-      }}
+      onClick={exhaustIfInteractive(onClick)}
     >
       <Icon glyph="Trash" />
     </IconButton>
   );
 };
+
+export const OpensInModal = ({ onClick, visible }: ActionButtonProps) => {
```

Comment:
For consistency with other action buttons

```suggestion
export const OpenBulkUpdateActionButton = ({ onClick, visible }: ActionButtonProps) => {
```